File: app/src/main/java/com/test/pokedex/network/models/pokemon_list/PokemonListResponse.java

package com.test.pokedex.network.models.pokemon_list;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class PokemonListResponse {

    @SerializedName("count")
    private Long mCount;
    @SerializedName("next")
    private String mNext;
    @SerializedName("previous")
    private Object mPrevious;
    @SerializedName("results")
    private List<Result> mResults;

    public Long getCount() {
        return mCount;
    }

    public void setCount(Long count) {
        mCount = count;
    }

    public String getNext() {
        return mNext;
    }

    public void setNext(String next) {
        mNext = next;
    }

    public Object getPrevious() {
        return mPrevious;
    }

    public void setPrevious(Object previous) {
        mPrevious = previous;
    }

    public List<Result> getResults() {
        return mResults;
    }

    public void setResults(List<Result> results) {
        mResults = results;
    }

}

File: app/src/main/java/com/test/pokedex/network/models/pokemon_list/PokemonListResponse.kt
package com.test.pokedex.network.models.pokemon_list

import com.google.gson.annotations.SerializedName
import com.test.pokedex.DataResponse

data class PokemonListResponse(
        @SerializedName("count")
        var count: Long? = null,

        @SerializedName("next")
        var next: String? = null,

        @SerializedName("previous")
        var previous: Any? = null,

        @SerializedName("results")
        var pokemonItems: ArrayList<PokemonItem?>? = null
) : DataResponse<ArrayList<PokemonItem?>?> {
    override fun retrieveData(): ArrayList<PokemonItem?>? {
        return pokemonItems
    }
}
